spec/enumerables_spec.rb

```diff
@@ -0,0 +1,129 @@
+require_relative '../enumerables.rb'
+
+describe Enumerable do
+  let(:array) { [1, 2, 3, 4, 5, 6, 7, 8, 9] }
+  let(:words_array) { %w[ant bear cat] }
+  let(:nil_true) { [nil, true, 99, false] }
+  let(:proced) { proc { |n| n * 2 } }
+  describe '#my_each' do
+    it 'Returns each element' do
+      expect(array.my_each { |n| n * n }).to eql(array)
+    end
+
+    it 'Returns enumerator' do
+      expect(array.my_each.class).to be(Enumerator)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'Returns each element' do
+      expect(array.my_each_with_index { |n| n * n }).to eql(array)
+    end
+
+    it 'Returns enumerator' do
+      expect(array.my_each_with_index.class).to be(Enumerator)
+    end
+  end
```

Comment:
You can also add a test to check whether it calls the given block or not by adding the following test,
```ruby
it 'calls the given block for each of the items in the array with its index' do
      each_output = 0
      block = proc { |val, idx| each_output += (val + idx) }
      array.my_each_with_index(&block)
      my_each_output = each_output.dup
      each_output = 0
      array.each_with_index(&block)
      expect(my_each_output).to eq(each_output)
end
```

Reply:
Done! Thank you for pointing it out! We added the test case!